### PR TITLE
Document use of transformers in Code component

### PR DESCRIPTION
**Pull request by matthewp**
#### Description (required)

Documents new `transformers` property on the Cod component.

#### Related issues & labels (optional)

- https://github.com/withastro/astro/pull/11197

### CODE REVIEW

**Comment by matthewp** [comment]
Short and sweet 😅. I wasn't sure what else to say about this.

File: src/content/docs/en/reference/api-reference.mdx [hunk 1]
```diff
@@ -1703,6 +1703,27 @@ import { Code } from 'astro:components';
...
+
+[Shiki transformers](https://shiki.style/packages/transformers#shikijs-transformers) can optionally be applied to code by passing them in through the `transformers` property as an array:
+
+```astro
+---
+import { transformerNotationFocus } from '@shikijs/transformers'
+import { Code } from 'astro:components'
+const code = `const foo = 'hello'
+const bar = ' world'
+console.log(hello + world) // [!code focus]
```
Comment (TheOtterlord) [inline src/content/docs/en/reference/api-reference.mdx:1718]:
```suggestion
console.log(foo + bar) // [!code focus]
```

**Review by TheOtterlord — COMMENTED** [review]
Copying one review comment from https://github.com/withastro/astro/pull/11197. The second comment doesn't apply because there's no css here

File: src/content/docs/en/reference/api-reference.mdx [hunk 2]
```diff
@@ -1703,6 +1703,27 @@ import { Code } from 'astro:components';
 
 This component provides syntax highlighting for code blocks at build time (no client-side JavaScript included). The component is powered internally by Shiki and it supports all popular [themes](https://shiki.style/themes) and [languages](https://shiki.style/languages). Plus, you can add your custom themes and languages by passing them to `theme` and `lang` respectively.
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:1704]:
```suggestion
This component provides syntax highlighting for code blocks at build time (no client-side JavaScript included). The component is powered internally by Shiki and it supports all popular [themes](https://shiki.style/themes) and [languages](https://shiki.style/languages). Plus, you can add your custom themes, languages, and [transformers](#transformers) by passing them to the `theme`, `lang`, and `transformers` attributes respectively.
```
Now that there's a third attribute, we should include it in the text above!

File: src/content/docs/en/reference/api-reference.mdx [hunk 3]
```diff
@@ -1703,6 +1703,27 @@ import { Code } from 'astro:components';
...
 
+#### Transformers
+
+<p><Since v="4.11.0" /></p>
+
+[Shiki transformers](https://shiki.style/packages/transformers#shikijs-transformers) can optionally be applied to code by passing them in through the `transformers` property as an array:
+
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:1711]:
```suggestion

Note that `transformers` only applies classes and you must provide your own CSS rules to target the elements of your code block.

```
This is the same suggestion I left in [withastro/astro/pull/11197](https://github.com/withastro/astro/pull/11197#discussion_r1644199714) -- something about needing to add your own CSS rules for styling.

File: src/content/docs/en/reference/api-reference.mdx [hunk 4]
```diff
@@ -1703,6 +1703,27 @@ import { Code } from 'astro:components';
...
+import { Code } from 'astro:components'
+const code = `const foo = 'hello'
+const bar = ' world'
+console.log(hello + world) // [!code focus]
+`
+---
+<Code
+  code={code}
+  lang="js"
+  transformers={[transformerNotationFocus()]} />
```
Comment (sarah11918) [inline src/content/docs/en/reference/api-reference.mdx:1724]:
```suggestion
  transformers={[transformerNotationFocus()]} />
  
  <style is:global>
    pre.has-focused .line:not(.focused) {
      filter: blur(1px);
    }
  </style>
```
With the extra mention about styles, I think showing a style tag is helpful, whatever makes sense.

**Review by sarah11918 — COMMENTED** [review]
Thank you @matthewp ! Just a couple of suggestions!

**Review by sarah11918 — APPROVED** [review]
Ready for 4.11! 🚀

**Pull request merged**